rust-toolchain

```diff
@@ -1 +1 @@
-1.51.0
+1.56.1
```

Comment:
Why is this change being made? I was specifically trying to avoid bumping our MSRV until after 4.7.0 (though I haven't succeeded; the circuit-change branch of `orchard` has MSRV 1.54 due to a mistake made when ZF published `reddsa 0.2.0`).

Reply:
Also, zcashd currently is using 1.56.1, so I assumed that that's what we'd be bumping to?

Reply:
Ah, I see this dependency was introduced 5 days ago, but because we have no MSRV-checking CI in the `incrementalmerkletree` crate, it went unnoticed. *sigh*

Reply:
I'll add CI there.